internal/bminventory/inventory.go

```diff
@@ -1911,9 +1936,81 @@ func (b *bareMetalInventory) validateAndUpdateClusterParams(ctx context.Context,
 		return installer.V2UpdateClusterParams{}, err
 	}
 
+	// We don't want to update ControlPlaneCount in day2 clusters as we don't know the previous hosts
+	if params.ClusterUpdateParams.ControlPlaneCount != nil && swag.StringValue(cluster.Kind) != models.ClusterKindAddHostsCluster {
+		if err := validateHighAvailabilityWithControlPlaneCount(
+			swag.StringValue(cluster.HighAvailabilityMode),
+			swag.Int64Value(params.ClusterUpdateParams.ControlPlaneCount),
+			cluster.OpenshiftVersion,
+		); err != nil {
+			return installer.V2UpdateClusterParams{}, err
+		}
+	}
+
 	return *params, nil
 }
 
+func validateHighAvailabilityWithControlPlaneCount(highAvailabilityMode string, controlPlaneCount int64, openshiftVersion string) error {
+	validControlPlaceCountValues := []int64{1, 3, 4, 5}
+	// Updating to invalid control plane count value
+	if !slices.Contains(validControlPlaceCountValues, controlPlaneCount) {
+		return common.NewApiError(
+			http.StatusBadRequest,
+			fmt.Errorf("valid values for control_plane_count are 1, 3, 4 or 5"),
+		)
+	}
+
+	// Updating SNO cluster to amount != 1
```

Comment:
small comment - we are not updating anything here 
```suggestion
	// validating the number of control planes for SNO 
```